Tap panic after a while

Unfortunately, I don't yet have a consistent repro, but I've hit this panic twice with `conduit tap` while sending lots of gRPC requests to a conduit-injected deployment:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1d0bb22]

goroutine 1 [running]:
github.com/runconduit/conduit/cli/cmd.print(0x285b500, 0xc42000a240)
	/go/src/github.com/runconduit/conduit/cli/cmd/tap.go:135 +0x9d2
github.com/runconduit/conduit/cli/cmd.glob..func6(0x28a4b40, 0xc4202ca980, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/runconduit/conduit/cli/cmd/tap.go:77 +0x2d5
github.com/runconduit/conduit/vendor/github.com/spf13/cobra.(*Command).execute(0x28a4b40, 0xc4202ca8e0, 0x2, 0x2, 0x28a4b40, 0xc4202ca8e0)
	/go/src/github.com/runconduit/conduit/vendor/github.com/spf13/cobra/command.go:698 +0x47a
github.com/runconduit/conduit/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x28a4700, 0x20725e0, 0xc42056ff70, 0x10043a4)
	/go/src/github.com/runconduit/conduit/vendor/github.com/spf13/cobra/command.go:783 +0x30e
github.com/runconduit/conduit/vendor/github.com/spf13/cobra.(*Command).Execute(0x28a4700, 0x1d57ddc, 0xc4202ca860)
	/go/src/github.com/runconduit/conduit/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/go/src/github.com/runconduit/conduit/cli/main.go:14 +0x31
```

I am using a custom-built Conduit version, which consists of master commit 81fb0fe onto which I have merged 22475d4 (from `alex/rbac` branch) and c31e892 (from `alex/skip-outbound-ports` branch).

I'm using the `conduit` CLI bin from the built image, and my `conduit install` and `conduit injects` are referencing my custom-built images.

--- a/cli/cmd/tap.go
+++ b/cli/cmd/tap.go
@@ -277,7 +277,7 @@ func renderTapEvent(event *common.TapEvent) string {
 			ev.ResponseInit.GetId().GetStream(),
 			flow,
 			ev.ResponseInit.GetHttpStatus(),
-			ev.ResponseInit.GetSinceRequestInit().Nanos/1000,
+			ev.ResponseInit.GetSinceRequestInit().GetNanos()/1000,
 			isSecured,
 		)
 

--- a/cli/cmd/tap.go
+++ b/cli/cmd/tap.go
@@ -259,8 +259,7 @@ func renderTapEvent(event *common.TapEvent) string {
 		dst,
 	)
 
-	httpEvent := event.GetHttp().GetEvent()
-	switch ev := httpEvent.(type) {
+	switch ev := event.GetHttp().GetEvent().(type) {
 	case *common.TapEvent_Http_RequestInit_:
 		return fmt.Sprintf("req id=%d:%d %s :method=%s :authority=%s :path=%s secured=%s",
 			ev.RequestInit.Id.Base,
@@ -283,44 +282,49 @@ func renderTapEvent(event *common.TapEvent) string {
 		)
 
 	case *common.TapEvent_Http_ResponseEnd_:
-		if ev.ResponseEnd.GetEos() != nil {
-			switch eos := ev.ResponseEnd.GetEos().GetEnd().(type) {
-			case *common.Eos_GrpcStatusCode:
-				return fmt.Sprintf("end id=%d:%d %s grpc-status=%s duration=%dµs response-length=%dB secured=%s",
-					ev.ResponseEnd.Id.Base,
-					ev.ResponseEnd.Id.Stream,
-					flow,
-					codes.Code(eos.GrpcStatusCode),
-					ev.ResponseEnd.GetSinceResponseInit().Nanos/1000,
-					ev.ResponseEnd.GetResponseBytes(),
-					isSecured,
-				)
-			case *common.Eos_ResetErrorCode:
-				return fmt.Sprintf("end id=%d:%d %s reset-error=%+v duration=%dµs response-length=%dB secured=%s",
-					ev.ResponseEnd.Id.Base,
-					ev.ResponseEnd.Id.Stream,
-					flow,
-					eos.ResetErrorCode,
-					ev.ResponseEnd.GetSinceResponseInit().Nanos/1000,
-					ev.ResponseEnd.GetResponseBytes(),
-					isSecured,
-				)
-			}
-		}
+		switch eos := ev.ResponseEnd.GetEos().GetEnd().(type) {
+		case *common.Eos_GrpcStatusCode:
+			return fmt.Sprintf("end id=%d:%d %s grpc-status=%s duration=%dµs response-length=%dB secured=%s",
+				ev.ResponseEnd.Id.Base,
+				ev.ResponseEnd.Id.Stream,
+				flow,
+				codes.Code(eos.GrpcStatusCode),
+				millisSinceInit(ev.ResponseEnd),
+				ev.ResponseEnd.GetResponseBytes(),
+				isSecured,
+			)
 
-		// this catchall handles 2 cases:
-		// 1) ev.ResponseEnd.Eos == nil
-		// 2) ev.ResponseEnd.Eos.End == nil
-		return fmt.Sprintf("end id=%d:%d %s duration=%dµs response-length=%dB secured=%s",
-			ev.ResponseEnd.Id.Base,
-			ev.ResponseEnd.Id.Stream,
-			flow,
-			ev.ResponseEnd.GetSinceResponseInit().Nanos/1000,
-			ev.ResponseEnd.GetResponseBytes(),
-			isSecured,
-		)
+		case *common.Eos_ResetErrorCode:
+			return fmt.Sprintf("end id=%d:%d %s reset-error=%+v duration=%dµs response-length=%dB secured=%s",
+				ev.ResponseEnd.Id.Base,
+				ev.ResponseEnd.Id.Stream,
+				flow,
+				eos.ResetErrorCode,
+				millisSinceInit(ev.ResponseEnd),
+				ev.ResponseEnd.GetResponseBytes(),
+				isSecured,
+			)
+
+		default:
+			return fmt.Sprintf("end id=%d:%d %s duration=%dµs response-length=%dB secured=%s",
+				ev.ResponseEnd.Id.Base,
+				ev.ResponseEnd.Id.Stream,
+				flow,
+				millisSinceInit(ev.ResponseEnd),
+				ev.ResponseEnd.GetResponseBytes(),
+				isSecured,
+			)
+		}
 
 	default:
 		return fmt.Sprintf("unknown %s", flow)
 	}
+}
+
+func millisSinceInit(end *common.TapEvent_Http_ResponseEnd) int32 {
+	var latency int32
+	if init := end.GetSinceResponseInit(); init != nil {
+		latency = init.Nanos / 1000
+	}
+	return latency
 }

--- a/cli/cmd/tap.go
+++ b/cli/cmd/tap.go
@@ -262,8 +262,8 @@ func renderTapEvent(event *common.TapEvent) string {
 	switch ev := event.GetHttp().GetEvent().(type) {
 	case *common.TapEvent_Http_RequestInit_:
 		return fmt.Sprintf("req id=%d:%d %s :method=%s :authority=%s :path=%s secured=%s",
-			ev.RequestInit.Id.Base,
-			ev.RequestInit.Id.Stream,
+			ev.RequestInit.GetId().GetBase(),
+			ev.RequestInit.GetId().GetStream(),
 			flow,
 			ev.RequestInit.GetMethod().GetRegistered().String(),
 			ev.RequestInit.GetAuthority(),
@@ -273,8 +273,8 @@ func renderTapEvent(event *common.TapEvent) string {
 
 	case *common.TapEvent_Http_ResponseInit_:
 		return fmt.Sprintf("rsp id=%d:%d %s :status=%d latency=%dµs secured=%s",
-			ev.ResponseInit.Id.Base,
-			ev.ResponseInit.Id.Stream,
+			ev.ResponseInit.GetId().GetBase(),
+			ev.ResponseInit.GetId().GetStream(),
 			flow,
 			ev.ResponseInit.GetHttpStatus(),
 			ev.ResponseInit.GetSinceRequestInit().Nanos/1000,
@@ -285,32 +285,32 @@ func renderTapEvent(event *common.TapEvent) string {
 		switch eos := ev.ResponseEnd.GetEos().GetEnd().(type) {
 		case *common.Eos_GrpcStatusCode:
 			return fmt.Sprintf("end id=%d:%d %s grpc-status=%s duration=%dµs response-length=%dB secured=%s",
-				ev.ResponseEnd.Id.Base,
-				ev.ResponseEnd.Id.Stream,
+				ev.ResponseEnd.GetId().GetBase(),
+				ev.ResponseEnd.GetId().GetStream(),
 				flow,
 				codes.Code(eos.GrpcStatusCode),
-				millisSinceInit(ev.ResponseEnd),
+				ev.ResponseEnd.GetSinceResponseInit().GetNanos()/1000,
 				ev.ResponseEnd.GetResponseBytes(),
 				isSecured,
 			)
 
 		case *common.Eos_ResetErrorCode:
 			return fmt.Sprintf("end id=%d:%d %s reset-error=%+v duration=%dµs response-length=%dB secured=%s",
-				ev.ResponseEnd.Id.Base,
-				ev.ResponseEnd.Id.Stream,
+				ev.ResponseEnd.GetId().GetBase(),
+				ev.ResponseEnd.GetId().GetStream(),
 				flow,
 				eos.ResetErrorCode,
-				millisSinceInit(ev.ResponseEnd),
+				ev.ResponseEnd.GetSinceResponseInit().GetNanos()/1000,
 				ev.ResponseEnd.GetResponseBytes(),
 				isSecured,
 			)
 
 		default:
 			return fmt.Sprintf("end id=%d:%d %s duration=%dµs response-length=%dB secured=%s",
-				ev.ResponseEnd.Id.Base,
-				ev.ResponseEnd.Id.Stream,
+				ev.ResponseEnd.GetId().GetBase(),
+				ev.ResponseEnd.GetId().GetStream(),
 				flow,
-				millisSinceInit(ev.ResponseEnd),
+				ev.ResponseEnd.GetSinceResponseInit().GetNanos()/1000,
 				ev.ResponseEnd.GetResponseBytes(),
 				isSecured,
 			)
@@ -319,12 +319,4 @@ func renderTapEvent(event *common.TapEvent) string {
 	default:
 		return fmt.Sprintf("unknown %s", flow)
 	}
-}
-
-func millisSinceInit(end *common.TapEvent_Http_ResponseEnd) int32 {
-	var latency int32
-	if init := end.GetSinceResponseInit(); init != nil {
-		latency = init.Nanos / 1000
-	}
-	return latency
 }

--- a/cli/cmd/tap.go
+++ b/cli/cmd/tap.go
@@ -259,19 +259,19 @@ func renderTapEvent(event *common.TapEvent) string {
 		dst,
 	)
 
-	http := event.GetHttp()
-	httpEvent := http.Event
+	httpEvent := event.GetHttp().GetEvent()
 	switch ev := httpEvent.(type) {
 	case *common.TapEvent_Http_RequestInit_:
 		return fmt.Sprintf("req id=%d:%d %s :method=%s :authority=%s :path=%s secured=%s",
 			ev.RequestInit.Id.Base,
 			ev.RequestInit.Id.Stream,
 			flow,
-			ev.RequestInit.Method.GetRegistered().String(),
-			ev.RequestInit.Authority,
-			ev.RequestInit.Path,
+			ev.RequestInit.GetMethod().GetRegistered().String(),
+			ev.RequestInit.GetAuthority(),
+			ev.RequestInit.GetPath(),
 			isSecured,
 		)
+
 	case *common.TapEvent_Http_ResponseInit_:
 		return fmt.Sprintf("rsp id=%d:%d %s :status=%d latency=%dµs secured=%s",
 			ev.ResponseInit.Id.Base,
@@ -281,10 +281,10 @@ func renderTapEvent(event *common.TapEvent) string {
 			ev.ResponseInit.GetSinceRequestInit().Nanos/1000,
 			isSecured,
 		)
-	case *common.TapEvent_Http_ResponseEnd_:
 
-		if ev.ResponseEnd.Eos != nil {
-			switch eos := ev.ResponseEnd.Eos.End.(type) {
+	case *common.TapEvent_Http_ResponseEnd_:
+		if ev.ResponseEnd.GetEos() != nil {
+			switch eos := ev.ResponseEnd.GetEos().GetEnd().(type) {
 			case *common.Eos_GrpcStatusCode:
 				return fmt.Sprintf("end id=%d:%d %s grpc-status=%s duration=%dµs response-length=%dB secured=%s",
 					ev.ResponseEnd.Id.Base,

